s3/stow_test.go

```diff
@@ -71,6 +71,39 @@ func TestPreSignedURL(t *testing.T) {
 	assert.NotEmpty(t, res)
 }
 
+func TestPreSignedURLSSEBucket(t *testing.T) {
+	is := is.New(t)
+	accessKeyId := os.Getenv("S3ACCESSKEYID")
+	secretKey := os.Getenv("S3SECRETKEY")
+	region := os.Getenv("S3REGION")
+
+	if accessKeyId == "" || secretKey == "" || region == "" {
+		t.Skip("skipping test because missing one or more of S3ACCESSKEYID S3SECRETKEY S3REGION")
+	}
+
+	config := stow.ConfigMap{
+		"access_key_id": accessKeyId,
+		"secret_key":    secretKey,
+		"region":        region,
+		"extra_args":    "{\"ServerSideEncryption\": \"aws:kms\", \"SSEKMSKeyId\": \"kmsId\"}",
```

Comment:
Would that be similar to this?
```
accessKeyId := os.Getenv("S3ACCESSKEYID")
secretKey := os.Getenv("S3SECRETKEY")
region := os.Getenv("S3REGION")
```

I think these are ENV vars because they are secret values – maybe I'm misunderstanding your comment, but I wouldn't test this function using ENV vars since I want to specify the particular input args and output URL strings.

Reply:
Yeah it was awkward to set it up with working S3 credentials (I hand-edited the test to work with my credentials) but I was able to test it locally, at least the pre-signed URL logic and checking headers as checked in this test case.